lib/src/rules/unnecessary_statements.dart

```diff
@@ -92,6 +94,48 @@ class _ReportNoClearEffectVisitor extends UnifyingAstVisitor {
 
   @override
   void visitBinaryExpression(BinaryExpression node) {
+    // look for the class of the left operand and its extensions
+    var clazz = context.currentUnit.unit.declarations
+        .whereType<ClassDeclaration>()
```

Comment:
Leaving open whether we agree that we *want* to loosen this check (which we should settle in the associated issue), I will note that if we do, we should try and optimized these new bits.

Here's a benchmark run before this change:

![image](https://github.com/dart-lang/linter/assets/67586/4dfd522b-8867-41c6-a225-8551ca6ef8fc)


And after:

![image](https://github.com/dart-lang/linter/assets/67586/50e1f042-8729-4c11-ae41-596211c9098f)

Either way, it does seem likely that we can establish if an operator is overriding more directly.  Maybe `UnnecessaryOverrides` can provide some ideas?  (In particular, some variation of `enclosingElement.thisType.lookUpMethod2(...)`?)